database/wrapper/player_wrapper.py

```diff
@@ -27,3 +28,31 @@ def get_random_player(session):
         logger.warning(e)
         player = create_default_player()
     return player
+
+
+class PlayerWrapper:
+    def __init__(self, limit=20):
+        self.limit = limit
+
+    def get_player_games(self, session, id, id_list=False):
+        if id_list:
+            # query = session.query(Game.hash).filter(cast(Game.players, postgresql.ARRAY(String)).contains([id1, id2]))
+            # query = session.query(Game.hash).filter(Game.players.op('@>')('{\'%s\', \'%s\'}' % (id1, id2)))
+            return session.query(PlayerGame).join(Game).filter(Game.players.contains(cast(id, postgresql.ARRAY(String)))).filter(
+                PlayerGame.player == id[0])
+        else:
+            return session.query(PlayerGame).filter(PlayerGame.player == id).filter(
+                PlayerGame.game is not None)
+
+    def get_player_games_paginated(self, session, id, id_list=False, page=0):
+        query = self.get_player_games(session, id, id_list=id_list)
+
+        return self.get_paginated_match_history(query, page=page, id_list=id_list)
+
+    def get_paginated_match_history(self, existing_query, page=0, id_list=False) -> List[PlayerGame]:
+        if not id_list:

```

Comment:
This seems really hacky, especially since we're already doing a join when we create `existing_query` so we shouldn't have to do it a second+ time.